### PR TITLE
Feature: Expose Devportal

### LINKED ISSUE #654: Expose Swagger UI from Kusk
Could we host swagger ui for the user and when they add this extension: 
```
x-kusk: 
  playground:
    path: /playground
```
We show them swagger UI under /playground (we could have a default path)

This way consumers can test the API, run operations, etc directly enabled by Kusk

### PR DESCRIPTION
**Pull request by jasmingacic**
Signed-off-by: jasmingacic <jasmin.gacic@gmail.com>

This PR addresses definition of DevPortal specs for backend as described here #654

closes #654 

## Changes

- Add manifests for spinning up devportal (deployment, service, envoyfleet, and staticroute)
- Allow devportal to be installed and upgraded via kusk

### CODE REVIEW

File: internal/controllers/parser.go [hunk 1]
```diff
@@ -459,6 +459,55 @@ func UpdateConfigFromAPIOpts(
 		}
 	}
 
+	if opts.DevPortal != nil {
+		for _, vh := range opts.Hosts {
+			//add devportal cluster
+			hostPortPair, err := getUpstreamHost(&options.UpstreamOptions{
+				Service: &options.UpstreamService{
+					Name:      "kusk-gateway-devportal",
```
Comment (kylehodgetts) [inline internal/controllers/parser.go:467]: We should get this from some config somewhere instead of hard coding it

**Comment by jasmingacic** [comment]
I can't approve this through GH since it was me who opened this PR. 
Consider it approved by me.
@mbana do  you want to chime in here?

**Comment by jasmingacic** [comment]
I tested the setup with images i built from this PR 
`jasmingacic/kusk-gateway:latest`
And from dashboard PR https://github.com/kubeshop/kusk-gateway-dashboard/pull/347
`jasmingacic/kusk-gateway-dashboard` 

it resulted in having this nice presentation 
image
image

**Review by mbana — COMMENTED** [review]
Just some general feedback whilst I try it out.

File: cmd/kusk/manifests/devportal.yaml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+          name: kusk-devportal
+          env:
+            - name: KUSK_GATEWAY_API_URL
+              value: "http://kusk-gateway-private-envoy-fleet.kusk-system.svc.cluster.local/api"
+          imagePullPolicy: Always
+          ports:
+            - containerPort: 80
+              name: http
+              protocol: TCP
+          resources:
```
[the remark this answers is not in the document]
Comment (kylehodgetts) [reply cmd/kusk/manifests/devportal.yaml:34]: For us, not really
but for a user who will install this, it's being a good citizen as it were in kubernetes

**Review by mbana — APPROVED** [review]
LGTM.

I was the logs arriving via a WebSocket to the dashboard.

**Pull request merged**